datafusion-cli/src/exec.rs

```diff
@@ -84,15 +84,20 @@ pub async fn exec_from_repl(ctx: &mut Context, print_options: &PrintOptions) {
     rl.set_helper(Some(CliHelper::default()));
     rl.load_history(".history").ok();
 
+    let mut print_options = print_options.clone();
+
     loop {
         match rl.readline("❯ ") {
             Ok(line) if line.starts_with('\\') => {
                 rl.add_history_entry(line.trim_end());
                 if let Ok(cmd) = &line[1..].parse::<Command>() {
                     match cmd {
                         Command::Quit => break,
+                        Command::QuietMode(quiet) => {
```

Comment:
I wonder why not move the execution to the command file? quit should be the only exception here?

Reply:
Fair question, but I did not dare to change the Command::execute signature (to make print_options mutable).
What do you think about that ?

Reply:
I don't believe it's used outside this repo nor any downstream code so you should feel free to change it as you see fit

Reply:
Ok then, changed to delegate the mutation to command